ns1/ipam.py

```diff
@@ -1050,12 +1053,45 @@ def update(self, address_family, options, server_options=None):
             self.server_options = server_options
 
         self.address_family = address_family
-        self.__dict__.update(options)
+        self.options = options
         self.__dict__.update(server_options)
+
         self.option_list = {
             "options": [
-                {"name": "%s/%s" % (self.address_family, key), "value": value}
-                for key, value in options.items()
+                option.generate_option(address_family)
+                for option in self.options
             ]
         }
+
         self.option_list.update(self.server_options)
+
+
+class DHCPOptionValue:
+    def __init__(self, key, value, always_send=None):
+        """
+        Create the DHCPOptionValue class that can be used as value with :class:`ns1.ipam.DHCPOptions`
+
+        :param key str option name
+        :param value any option value
+        :param always_send bool indicates whether this option be sent back in lease or not
+        """
+        self.key = key
+        self.value = value
+        self.always_send = (
+            bool(always_send) if always_send is not None else always_send
+        )
+
+    def generate_option(self, address_family):
+        """
+        Generates dhcp option value with a proper format
+
+        :param str address_family one of dhcpv4 or dhcpv6 family name
```

Comment:
since there's only two choices for address_family, would some validation be appropriate here?

Reply:
I think we have such checks on becked. Any way we can add this check, but what will the action if address_family will be corrupted? Rise an exception or do nothing and add nothing?

Reply:
that's why it's a question :) - ideally we'd raise a ValueError, but if there isn't a good way to recover or get the message to the user in the use-cases we expect, and if it'll get validated later, it might not be worth it

Reply:
agree )

Reply:
Returning the error from the API might suffice here.

Reply:
Definitely response from the API will hold such err - there is a related check on the back end for this.

Reply:
The user will have an API response if an error occurs.